module_scripts.py

```diff
@@ -14619,6 +14615,46 @@ def animation_menu_entry(string_id, **kwargs):
         (try_end),
       (try_end),
     ]),
+    
+    ("check_wielding_while_sitting", [
+      (store_script_param_1, ":agent_id"),
+      (store_script_param_2, ":item_id"),
+      (try_begin),
+        (multiplayer_is_server),
+        (agent_is_active, ":agent_id"),
+        (agent_is_alive, ":agent_id"),
+        (agent_slot_eq, ":agent_id", slot_agent_is_in_position_animation, 1),
+        (try_begin),
+          (set_fixed_point_multiplier, 100),
+            
+          (agent_get_position, pos10, ":agent_id"),
+            
+          (agent_get_slot, ":x", ":agent_id", slot_agent_position_animation_x),
+          (agent_get_slot, ":y", ":agent_id", slot_agent_position_animation_y),
+          (position_set_x, pos11, ":x"),
+          (position_set_y, pos11, ":y"),
+            
+          (position_set_z_to_ground_level, pos10),
+          (position_set_z_to_ground_level, pos11),
+            
+          (get_distance_between_positions, ":distance", pos10, pos11),
+          (le, ":distance", 15),

```

Comment:
Are the agent position slots even needed anymore? If the animation cancels when you move can you not simply find what animation the player is doing and then just assume that these checks need to be run?